snuba/datasets/errors.py

```diff
@@ -252,4 +263,4 @@ def get_extensions(self) -> Mapping[str, QueryExtension]:
         }
 
     def get_query_processors(self) -> Sequence[QueryProcessor]:
-        return [BasicFunctionsProcessor(), PrewhereProcessor()]
+        return [BasicFunctionsProcessor()]
```

Comment:
Why do some parts of query processing move to the query plan builder? Does this mean we are now flipping the order of these processors? (assuming that query plan builder runs first)

Reply:
There are three phases in query processing:
1) dataset query processing. This stays in the dataset. It happens based on the snuba dataset query before selecting the storage. It is based on an abstract schema, not on the clickhouse columns (thus basicFunctions and the like)
2) Storage specific query processing. These processor are specific to the table we are querying. Imagine FlattenedTags.  They are obviously provided by the Storage class and they are supposed to be executable no matter if the Storage is queries in isolation or as a join.
3) Post processors. These are per query executed on the storage based query after the storage ones. They do not depend on the storage itself, but just on the query after the plan has been created (like Sampling and prewhere). They need the context of the final query before formatting, which tells for example what we are joining with what. 

There was no reshuffling in plan, the PRewhere was before the FlattenedTags by mistake but the order of those two does not matter, they are independent.